### PR TITLE
Add exception continuation handler

### PR DESCRIPTION
**Pull request by jhand2**
This takes the exception continuation feature added by @mikbras in the lsds fork of OE (https://github.com/lsds/openenclave) which allows exceptions thrown inside the enclave to be handled and potentially allow execution to continue. The enclave writer simple writes a hook function that will be called in the event of an exception.

The goal of this PR is to review the design and discuss alternatives.

### CODE REVIEW

**Comment by jhand2** [comment]
In general I would prefer this behave more like the `signal` and `sigaction` Linux syscalls: http://man7.org/linux/man-pages/man2/sigaction.2.html

Given that we try to replicate a Linux syscall infrastructure inside the enclave, I think that interface would feel more ergonomic.

**Comment by jhand2** [comment]
The cpuid test is a bit more involved than is actually needed to prove that the exception handler is working. Need to come up with a more minimal test.

File: tests/cpuid/enc/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+
+oeedl_file(../cpuid.edl enclave gen)
```
Comment (mingweishih) [inline tests/cpuid/enc/CMakeLists.txt]: Note that `oeedl_file` is deprecated (see #2325).

File: tests/cpuid/host/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+
+oeedl_file(../cpuid.edl host gen)
```
Comment (mingweishih) [inline tests/cpuid/host/CMakeLists.txt]: oeedl_file is deprecated.

File: tests/cpuid/enc/enc.c [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    }
+}
+
+OE_SET_ENCLAVE_SGX(
+    1,    /* ProductID */
+    1,    /* SecurityVersion */
+    true, /* AllowDebug */
+    1024, /* HeapPageCount */
+    1024, /* StackPageCount */
+    2);   /* TCSCount */
```
Comment (mingweishih) [inline tests/cpuid/enc/enc.c]: Does the test require TCSCount =2?

**Comment by jhand2** [comment]
After more investigation I'm not convinced this PR is necessary. Closing for now.

**Pull request closed without merging**